File: mobile/app/src/main/java/com/praeter/ui/base/BaseFragment.java
package com.praeter.ui.base;

import android.os.Bundle;

import androidx.annotation.Nullable;

import javax.inject.Inject;

import dagger.android.support.DaggerFragment;

public class BaseFragment<V extends BaseView> extends DaggerFragment {

    @Inject
    public V view;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        view.onCreate();
    }

    @Override
    public void onDetach() {
        view.onDestroy();
        super.onDetach();
    }
}

File: mobile/app/src/main/java/com/praeter/ui/base/BaseFragment.kt
package com.praeter.ui.base

import android.os.Bundle
import androidx.fragment.app.Fragment
import dagger.hilt.android.AndroidEntryPoint
import timber.log.Timber

@AndroidEntryPoint
open class BaseFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Timber.d("onCreate()")
    }

    override fun onDestroy() {
        super.onDestroy()
        Timber.e("onDestroy()")
    }

}